rosetta/translate_utils.py

```diff
@@ -0,0 +1,114 @@
+    https://docs.microsoft.com/en-us/azure/cognitive-services/translator/reference/v3-0-translate?tabs=curl
+    """
+
+    AZURE_TRANSLATOR_HOST = 'https://api.cognitive.microsofttranslator.com'
+    AZURE_TRANSLATOR_PATH = '/translate?api-version=3.0'
+
+    headers = {
+        'Ocp-Apim-Subscription-Key': subscription_key,
+        'Content-type': 'application/json',
+        'X-ClientTraceId': str(uuid.uuid4())
+    }
+
+    url_parameters = {
+        "from": from_language,
+        "to": to_language
+    }
+
+    request_data = [
+        {"text": text}
+    ]
+
+    api_hostname = AZURE_TRANSLATOR_HOST + AZURE_TRANSLATOR_PATH
+    r = requests.post(api_hostname, headers=headers, params=url_parameters, data=json.dumps(request_data))
+    api_response = json.loads(r.text)
+
+    try:
+        # result will be a dict if there is an error, e.g.
+        # {
+        #   "success": false,
+        #    "error": "Microsoft Translation API error: Error code 401000,
+        #             The request is not authorized because credentials are missing or invalid."
+        # }
+        if isinstance(api_response, dict):
+            api_error = api_response.get("error")
+            error_code = api_error.get("code")
+            error_message = api_error.get("message")
+
+            raise TranslationException("Microsoft Translation API error: Error code {}, {}".format(error_code, error_message))
+        else:
+            # response body will be of the form:
+            # [
+            #     {
+            #         "translations":[
+            #             {"text": "some chinese text that gave a build error on travis ci", "to": "zh-Hans"}
+            #         ]
+            #     }
+            # ]
+            # for more information, please visit
+            # https://docs.microsoft.com/en-us/azure/cognitive-services/translator/reference/v3-0-translate?tabs=curl
+
+            translations = api_response[0].get("translations")
+            translated_text = translations[0].get("text")
+
+            return translated_text
+    except requests.exceptions.RequestException as err:
+        raise TranslationException("Error connecting to Microsoft Translation Service: {0}".format(err))
+
+
+def translate_by_google(text, input_language, output_language, creadentials_path, project_id):
+    from google.cloud import translate_v3beta1 as google_translate
+
+    # using new v3beta Translation API
+    client = google_translate.TranslationServiceClient.from_service_account_json(creadentials_path)
+    parent = client.location_path(project_id, 'global')
+
+    try:
+        api_response = client.translate_text(
+            parent=parent,
+            contents=[text],
+            mime_type='text/plain',
+            source_language_code=input_language,
+            target_language_code=output_language
+        )
+    except Exception as e:
+        raise TranslationException('Google API error: {}'.format(e))
+    else:
```

Comment:
Can we simplify this code? The `else` is likely unnecessary here. While this is technically correct many developers are confused by it. There is no change in the business logic if we omit the `else`, is it?